test/jdk/java/awt/Focus/QuickTypeTest.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2002, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+  @test
+  @bug 4423838
+  @summary KEY_TYPED and KEY_PRESSED generated by the same key are notified to different fi
+  @key headful
+  @requires (os.family == "windows")
+  @run main QuickTypeTest
+*/
+
+import java.awt.AWTException;
+import java.awt.BorderLayout;
+import java.awt.Dialog;
+import java.awt.Dimension;
+import java.awt.Frame;
+import java.awt.Panel;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.TextArea;
+import java.awt.EventQueue;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+import java.util.Properties;
+
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+
+public class QuickTypeTest {
+    static final int TEST_TIMEOUT=10000;
+    static JFrame frame1;
+    static JFrame frame2;
+    static JTextField tf1;
+    static JTextField tf2;
+    static SmartKeyAdapter ska;
+    static Object keyMonitor;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                // Check for Win32 environment. This test is only for Win32.
+                Properties prop = System.getProperties();
```

Comment:
Is it really ?

Reply:
Took the test as it is..maybe at the time of test being wriiten, there was some issue with other platforms..
seems to be working on mach5 headful platforms, so remove the windows restriction